ui/pages/swaps/awaiting-signatures/awaiting-signatures.js

```diff
@@ -0,0 +1,98 @@
+import React, { useContext, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+
+import { I18nContext } from '../../../contexts/i18n';
+import { useNewMetricEvent } from '../../../hooks/useMetricEvent';
+import {
+  getFetchParams,
+  getApproveTxParams,
+  navigateBackToBuildQuote,
+} from '../../../ducks/swaps/swaps';
+import PulseLoader from '../../../components/ui/pulse-loader';
+import SwapsFooter from '../swaps-footer';
+import SwapStepIcon from './swap-step-icon';
+
+export default function AwaitingSignatures() {
+  const t = useContext(I18nContext);
+  const history = useHistory();
+  const dispatch = useDispatch();
+  const fetchParams = useSelector(getFetchParams);
+  const { destinationTokenInfo, sourceTokenInfo } = fetchParams?.metaData || {};
+  const approveTxParams = useSelector(getApproveTxParams);
+  const needsTwoConfirmations = Boolean(approveTxParams);
+
+  const awaitingSignaturesEvent = useNewMetricEvent({
+    event: 'Awaiting Signature(s) on a HW wallet',
+    sensitiveProperties: {
+      needs_two_confirmations: needsTwoConfirmations,
+      token_from: sourceTokenInfo?.symbol,
+      token_from_amount: fetchParams?.value,
+      token_to: destinationTokenInfo?.symbol,
+      request_type: fetchParams?.balanceError ? 'Quote' : 'Order',
+      slippage: fetchParams?.slippage,
+      custom_slippage: fetchParams?.slippage === 2,
+    },
+    category: 'swaps',
+  });
+
+  useEffect(() => {
+    awaitingSignaturesEvent();
+  }, [awaitingSignaturesEvent]);
+
+  const headerText = needsTwoConfirmations
+    ? t('swapTwoTransactions')
+    : t('swapConfirmWithHwWallet');
+
+  return (
+    <div className="awaiting-signatures">
+      <div className="awaiting-signatures__content">
+        <div className="awaiting-signatures__status-image">
+          <PulseLoader />
+        </div>
+        <div className="awaiting-signatures__header">{headerText}</div>
+        {needsTwoConfirmations && (
+          <>
+            <p className="awaiting-signatures__subheader">
```

Comment:
Could we make use of the Typography components to reduce the amount of new scss we introduce? I know this is a pre-existing pattern even in the swaps feature but I'm really pushing for adoption of minimizing styles where we have Typography/Box etc components to help

Reply:
That sounds great. I'm actually used to do that while using a framework like React-Bootstrap or Material UI. I used this approach just because I saw something similar on other pages. Let's sync up tomorrow!